helpers/sqs_helper.py

```diff
@@ -0,0 +1,108 @@
+"""
+Helper module for sqs messages
+"""
+import json
+import os
+import sys
+import time
+import boto3
+from botocore.exceptions import ClientError
+
+# add project root to sys path
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+from helpers.base_helper import BaseHelper
+from conf import sqs_conf
+
+class SqsHelper(BaseHelper):
+    """
+    SQS Helper object
+    """
+
+    def get_sqs_client(self):
+        """
+        Return sqs_client object
+        :param self:
+        :return sqs_client: SQS client object
+        """
+        try:
+            sqs_client = boto3.client('sqs', config=sqs_conf.config)
+            self.write(f'Created SQS client')
+        except ClientError as err:
+            self.write(f'Exception - {err}, Unable to create SQS client', level='error')
+        except Exception as err:
+            self.write(f'Unable to create SQS client, due to {err}', level='error') 
+
+        return sqs_client
+
+    def get_sqs_queue(self, queue_name):
+        """
+        Get queue
+        :param self:
+        :param queue_name: queue name
+        :return queue: queue object
+        """
+        try:
+            queue = boto3.resource('sqs').get_queue_by_name(QueueName=queue_name)
+            self.write(f'Attained SQS queue by name')
+        except ClientError as err:
+            self.write(f'Exception - {err}, Unable to get SQS queue', level='error')
+        except Exception as err:
+            self.write(f'Unable to get SQS queue, due to {err}', level='error') 
+
+        return queue
+
+    def get_message_from_queue(self, queue_name, attempts=3):
+        """
+        Get message from queue
+        :param self:
+        :param queue_name: queue name
+        :param attempts: number of attempts to get the message
+        :return messages: messages list object
+        """
+        try:
+            sqs_client = self.get_sqs_client()
+            queue = self.get_sqs_queue(queue_name)
```

Comment:
I saw this pattern in Rahul's code too. This makes not sense to me. Perhaps I am missing something obvious.

Why do we need a `queue` object here when we have a client? You are using the `queue` variable is to pass the `QueueUrl` to `receive_message`. However, `QueueUrl` is just a string ... so you did not need this `queue` object at all.

Reply:
Ignore me. I assumed Queue URL and queue name are the same. Silly me.